tests/rptest/tests/topic_delete_test.py

```diff
@@ -66,3 +69,65 @@ def topic_storage_purged():
                     self.logger.error(line.strip())
 
             raise
+
+
+class TopicDeleteStressTest(RedpandaTest):
+    def __init__(self, test_context):
+        extra_rp_conf = dict(
+            log_segment_size=1048576,
+            log_compaction_interval_ms=300,
+            auto_create_topics_enabled=False,
+        )
+
+        super(TopicDeleteStressTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=3,
+                             extra_rp_conf=extra_rp_conf)
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.context = test_context
+
+    @cluster(num_nodes=4)
+    def stress_test(self):
+        for i in range(10):
+            spec = TopicSpec(partition_count=2,
+                             cleanup_policy=TopicSpec.CLEANUP_COMPACT)
+            topic_name = spec.name
+            self.kafka_tools.create_topic(spec)
+
+            producer = KafProducer(self.context, self.redpanda, topic_name,
+                                   100000)
+            producer.start()
+
+            time.sleep(30)
```

Comment:
prefer waiting on a target system state. for example, you could wait for 10 segments to have been created and compaction having started on at least one. most of this information is available through the metrics api.